### ISSUE TITLE
Adding anchor AppProperties crashes in Android NDK

### ISSUE THREAD

**Issue by sergeiromanov** [Azure/azure-spatial-anchors-samples#297]
https://github.com/Azure/azure-spatial-anchors-samples/blob/79dc2ee96d027f9eeeab56767b5013d533bced63/Android/NDK/app/src/main/cpp/AzureSpatialAnchorsApplication.cpp#L570

Setting AppProperties will crash the program. I added this code:

```
visual.cloudAnchor = std::make_shared<CloudSpatialAnchor>();
visual.cloudAnchor->LocalAnchor(visual.localAnchor);
auto properties = visual.cloudAnchor->AppProperties(); // verified properties is not null
properties->Insert(R"(model-type)", R"(frame)");       // crash!
ERROR: A/libc: Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0 in tid 22056
```

Is it a bug or am I doing it wrong?

**Comment by pamistel** [comment]
Thank you for drawing our attention to this. We will attempt to repro and get back to you in this thread

**Comment by pamistel** [comment]
Hi @sergeiromanov 
Can you try it using 
`properties->Insert(LR"(model-type)", LR"(frame)"); `

**Comment by sergeiromanov** [comment]
Thanks for looking into this issue. The suggested code doesn't compile.
`error: reference to type 'const std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> >' could not bind to an lvalue of type 'const wchar_t [11]'`
Passing std::string crashes. Here is a backtrace:
```
#00 pc 000000000004ae00  /apex/com.android.runtime/lib64/bionic/libc.so (strlen_default+16) (BuildId: 47fa1511afbec3da8219bfec53725f93)
#01 pc 00000000000b6958  /data/app/~~nqCWgkY9JcApILlubq0AaA==/com.microsoft.samplenativeandroid-xa1NLKIRMn3rTe1ybDouuw==/base.apk!libazurespatialanchorsapplication.so (offset 0xc72000) (std::__ndk1::char_traits<char>::length(char const*)+20) (BuildId: cb457269776e70099d03020819279f7b6421e765)
#02 pc 00000000000a3a58  /data/app/~~nqCWgkY9JcApILlubq0AaA==/com.microsoft.samplenativeandroid-xa1NLKIRMn3rTe1ybDouuw==/base.apk!libazurespatialanchorsapplication.so (offset 0xc72000) (std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> >::basic_string<std::nullptr_t>(char const*)+48) (BuildId: cb457269776e70099d03020819279f7b6421e765)
#03 pc 00000000000dfc14  /data/app/~~nqCWgkY9JcApILlubq0AaA==/com.microsoft.samplenativeandroid-xa1NLKIRMn3rTe1ybDouuw==/base.apk!libazurespatialanchorsapplication.so (offset 0xc72000) (Microsoft::ApiGen::IDictionary_String_String::HasKey(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> > const&) const+112) (BuildId: cb457269776e70099d03020819279f7b6421e765)
#04 pc 00000000000dfcb0  /data/app/~~nqCWgkY9JcApILlubq0AaA==/com.microsoft.samplenativeandroid-xa1NLKIRMn3rTe1ybDouuw==/base.apk!libazurespatialanchorsapplication.so (offset 0xc72000) (Microsoft::ApiGen::IDictionary_String_String::Insert(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> > const&, std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char> > const&)+48) (BuildId: cb457269776e70099d03020819279f7b6421e765)
#05 pc 00000000000aeebc  /data/app/~~nqCWgkY9JcApILlubq0AaA==/com.microsoft.samplenativeandroid-xa1NLKIRMn3rTe1ybDouuw==/base.apk!libazurespatialanchorsapplication.so (offset 0xc72000) (AzureSpatialAnchors::AzureSpatialAnchorsApplication::CreateCloudAnchor()+300) (BuildId: cb457269776e70099d03020819279f7b6421e765)
```

**Comment by pamistel** [comment]
Hi @sergeiromanov

Thank you again for bringing our attention to this. We have created a fix for this and will be published with the next SDK release

**Comment by sergeiromanov** [comment]
Hi, thanks for looking into the issue. When is the next release scheduled? And where to watch it?

**Comment by msftradford** [comment]
Hi @sergeiromanov! The issue (#3359432) will be fixed as a part of the 2.11.0 release. We roughly estimate to have the fix released by end of next week 11/19/21. If you click the button to "watch" this repo then you will receive an email automatically when the release occurs. Otherwise, keep track of the releases on [this page](https://github.com/Azure/azure-spatial-anchors-samples/releases).
